File: backend/src/main/java/com/davnig/invoicemanagementapi/model/entity/InvoiceItem.java
package com.davnig.invoicemanagementapi.model.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class InvoiceItem {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private String description;
    private int price;
    private int quantity;

}

File: backend/src/main/kotlin/com/davnig/invoicemanagementapi/model/entity/InvoiceItem.kt
package com.davnig.invoicemanagementapi.model.entity

import jakarta.persistence.*

@Entity
@Table(name = "t_invoice_item")
class InvoiceItem {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "invoice-item-id-gen")
    @SequenceGenerator(name = "invoice-item-id-gen", sequenceName = "t_invoice_item_id_seq", allocationSize = 1)
    var id = 0

    @ManyToOne
    var type: InvoiceItemType? = null
    var description: String? = null
    var price = 0
    var quantity = 0

}